### DISCUSSION TITLE
Livewire doesn't work with select input `selected` attribute

### DISCUSSION THREAD

**Discussion by wojciechczerniak** [livewire/livewire#5081] [7. Bugs (archived)]
### Problem

If Laravel returns updated markup for `<select>` input, Livewire will try to morph the DOM and fails for some reason.

Problem is already described in https://github.com/livewire/livewire/discussions/2992 but while mutating values is a nice feature, it may be considered non essential. Input `selected` state tho is a built-in web platform feature that should work.

```blade
  <select wire:model="state.marital_status">
    <option value="" @if($state['marital_status'] == '') selected @endif disabled>Select one</option>
    <option value="married" @if($state['marital_status'] == 'married') selected @endif >Married</option>
    <option value="single" @if($state['marital_status'] == 'single') selected @endif >Single</option>

    <option value="divorced" @selected($state['marital_status'] == 'divorced') >Divorced</option>
  </select>
```

Playground: https://laravelplayground.com/#/snippets/164eff58-63c0-46ce-b84d-074e6d685b48

Notice that when you change an option it goes blank after a second

### Remove `selected` "solution"

I've seen in few threads people suggesting to remove `selected` attribute and leaving it up to Livewire to handle the state. Which IMO is a really bad idea, and not a solution at all. It will break the form for non-JS user agents, and cripple a perfectly valid HTML output generated with Laravel. I see Livewire as a progressive way to enhance UX. Without it, exactly the same valid Laravel output should work as intended. So each time a page is reloaded the `selected` attribute should be there.

**Comment by Martleby** [comment]
You could shove a wire:ignore on the select. I don't know if that would have knock-on effects...

Cheers,
Martin.

**Reply by wojciechczerniak** [reply]
Thank you @Martleby for taking time to answer, that's a workaround I'm currently using. I haven't noticed any undesired effects. `<select>` behaves just like ordinary select element, preserving attribute and updating property only.

Although, that does not solve the issue with DOM diffing

**Comment by joshhanley** [comment]
@wojciechczerniak yeah there was a DOM diffing issue with the displayed selected option. I've fixed this issue in PR #5129. Can you test it out and let us know how it goes? Thanks!

**Comment by samuelaustinudhedhe** [comment]
Hello @wojciechczerniak 
I have build a component that works with Alpine.js to fix the selected attribute issue when using select element with Livewire.

My Laravel Livewire Select Component is a custom component designed to enhance the functionality of HTML elements when used with Laravel Livewire. This component seamlessly integrates with Livewire to manage the default selected state of options, ensuring compatibility with Livewire's reactivity. It allows for programmatic default option selection and optimizes performance through an optional JavaScript loading feature.

`  <x-select wire:model="state.marital_status" loadJS="true">
    <option value="" @if($state['marital_status'] == '') selected @endif disabled>Select one</option>
    <option value="married" @if($state['marital_status'] == 'married') selected @endif >Married</option>
    <option value="single" @if($state['marital_status'] == 'single') selected @endif >Single</option>
    <option value="divorced" @selected($state['marital_status'] == 'divorced') >Divorced</option>
  </x-select> `

You can fine the component an instruction on how to use it here:

https://github.com/samuelaustinudhedhe/Laravel-Livewire-Select-Component

Let me know if you need further assistance
